java-checks/src/main/java/org/sonar/java/checks/ObjectCreatedOnlyToCallGetClassCheck.java

```diff
@@ -0,0 +1,119 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.java.checks;
+
+import com.google.common.collect.ImmutableList;
+import org.sonar.api.server.rule.RulesDefinition;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.java.model.AbstractTypedTree;
+import org.sonar.java.resolve.Symbol;
+import org.sonar.java.resolve.Symbol.TypeSymbol;
+import org.sonar.java.resolve.Type;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.MemberSelectExpressionTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.Tree.Kind;
+import org.sonar.plugins.java.api.tree.VariableTree;
+import org.sonar.squidbridge.annotations.ActivatedByDefault;
+import org.sonar.squidbridge.annotations.SqaleConstantRemediation;
+import org.sonar.squidbridge.annotations.SqaleSubCharacteristic;
+
+import java.util.List;
+
+@Rule(
+  key = "S2133",
+  name = "Objects should not be created only to \"getClass\"",
+  tags = {"performance"},
+  priority = Priority.BLOCKER)
+@ActivatedByDefault
+@SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.FAULT_TOLERANCE)
+@SqaleConstantRemediation("2min")
+public class ObjectCreatedOnlyToCallGetClassCheck extends SubscriptionBaseVisitor {
+
+  @Override
+  public List<Kind> nodesToVisit() {
+    return ImmutableList.of(Kind.METHOD_INVOCATION);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    MethodInvocationTree mit = (MethodInvocationTree) tree;
+    if (hasSemantic() && mit.methodSelect().is(Kind.MEMBER_SELECT)) {
+      ExpressionTree expressionTree = ((MemberSelectExpressionTree) mit.methodSelect()).expression();
+      boolean calledFromNewObject = expressionTree.is(Kind.NEW_CLASS);
+      boolean calledFromVariable = expressionTree.is(Kind.IDENTIFIER);
+      if (isGetClassInvocation(mit) && (calledFromNewObject || calledFromVariable)) {
+        Tree issueTarget = null;
+        if (calledFromVariable && isInstantiationOnlyOtherOperation(expressionTree)) {
+          issueTarget = getInstantiationTree(expressionTree);
+        } else if (calledFromNewObject) {
+          issueTarget = expressionTree;
+        }
+        if (issueTarget != null) {
+          addIssue(issueTarget, "Remove this object instantiation and use \"" + getTypeName(issueTarget) + ".class\" instead.");
+        }
+      }
+    }
+  }
+
+  private String getTypeName(Tree tree) {
+    Type type = ((AbstractTypedTree) tree).getSymbolType();
+    TypeSymbol symbol = type.getSymbol();
```

Comment:
To allow further reuse I would split this into a method getting a name from TypeSymbol.